tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/util/TargetPlatformUtil.java

```diff
@@ -0,0 +1,159 @@
+import java.util.List;
+import java.util.Map;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.eclipse.core.runtime.FileLocator;
+import org.eclipse.core.runtime.Platform;
+
+import com.liferay.ide.project.core.ITargetPlatformConstant;
+import com.liferay.ide.project.core.ProjectCore;
+
+/**
+ * @author Lovett Li
+ */
+public class TargetPlatformUtil
+{
+
+    public static String[] getServicesList() throws Exception
+    {
+        File tpIndexFile = checkCurrentTargetPlatform( "service" );
+
+        return getServicesNameList( tpIndexFile );
+    }
+
+    public static String[] getServiceBundle( String serviceName ) throws Exception
+    {
+        File tpIndexFile = checkCurrentTargetPlatform( "service" );
+
+        return getBundleAndVersion( tpIndexFile , serviceName);
+    }
+
+    public static String[] getServiceWrapperList() throws Exception
+    {
+
+        File tpIndexFile = checkCurrentTargetPlatform( "servicewrapper" );
+
+        return getServicesNameList( tpIndexFile );
+    }
+
+    public static String[] getServiceWrapperBundle( String servicewrapperName ) throws Exception
+    {
+
+        File tpIndexFile = checkCurrentTargetPlatform( "servicewrapper" );
+
+        return getBundleAndVersion( tpIndexFile , servicewrapperName);
+    }
+
+    private static File checkCurrentTargetPlatform( String type ) throws IOException
+    {
+        String currentVersion = Platform.getPreferencesService().getString(
+            ProjectCore.PLUGIN_ID, ITargetPlatformConstant.CURRENT_TARGETFORM_VERSION,
+            ITargetPlatformConstant.DEFAULT_TARGETFORM_VERSION, null );
+
+        currentVersion = currentVersion.replace( "[", "" ).replace( "]", "" ).toLowerCase();
+
+        return useSpecificTargetPlatform( currentVersion, type );
+    }
+
+    private static File useSpecificTargetPlatform( String currentVersion, String type ) throws IOException
+    {
+        URL url;
+        url = FileLocator.toFileURL(
+            ProjectCore.getDefault().getBundle().getEntry( "OSGI-INF/target-platform/liferay-" + currentVersion ) );
+        final File tpFolder = new File( url.getFile() );
+
+        File[] listFiles = tpFolder.listFiles( new FilenameFilter()
+        {
+
+            @Override
+            public boolean accept( File dir, String name )
+            {
+                if( type.equals( "service" ) && name.endsWith( "services.json" ) )
+                {
+                    return true;
+                }
+                if( type.equals( "servicewrapper" ) && name.endsWith( "servicewrappers.json" ) )
+                {
+                    return true;
+                }
+                return false;
+            }
+        } );
+
+        return listFiles[0];
+
+    }
+
+    public static List<String> getAllTargetPlatfromVersion() throws IOException
+    {
+        final URL url =
+            FileLocator.toFileURL( ProjectCore.getDefault().getBundle().getEntry( "OSGI-INF/target-platform" ) );
```

Comment:
Inside this target-platform we should use versions for the directory names.
liferay-ce-7.0.0-ga2
liferay-dxp-7.0.10-ga1, etc